golang/main.go

```diff
@@ -0,0 +1,183 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	_ "github.com/lib/pq"
+	"github.com/twmb/franz-go/pkg/kgo"
+)
+
+const batchSize = 1000
+
+type Event struct {
+	OrganizationID int                    `json:"organization_id,omitempty"`
+	CustomerID     *int                   `json:"customer_id,omitempty"`
+	CustID         string                 `json:"cust_id,omitempty"`
+	EventName      string                 `json:"event_name,omitempty"`
+	TimeCreated    time.Time              `json:"time_created,omitempty"`
+	Properties     map[string]interface{} `json:"properties,omitempty"`
+	IdempotencyID  string                 `json:"idempotency_id,omitempty"`
+	InsertedAt     time.Time              `json:"inserted_at,omitempty"`
+}
+
+type StreamEvents struct {
+	Events         []Event `json:"events"`
+	OrganizationID int64   `json:"organization_id"`
+	Event          *Event  `json:"event"`
+}
+
+type batch struct {
+	tx         *sql.Tx
+	insertStmt *sql.Stmt
+	count      int
+}
+
+func (b *batch) addRecord(event *Event) error {
+	propertiesJSON, errJSON := json.Marshal(event.Properties)
+	if errJSON != nil {
+		fmt.Printf("Error encoding properties to JSON: %s\n", errJSON)
+		return errJSON
+	}
+
+	_, err := b.insertStmt.Exec(event.OrganizationID, event.CustomerID, event.CustID, event.EventName, event.TimeCreated, propertiesJSON, event.IdempotencyID, event.InsertedAt)
+	if err != nil {
+		return err
+	}
+
+	b.count++
+	if b.count >= batchSize {
+		if err := b.tx.Commit(); err != nil {
+			return err
+		}
+		b.count = 0
+	}
+
+	return nil
+}
+
+func main() {
+	var kafkaURL string
+	if kafkaURL = os.Getenv("KAFKA_URL"); kafkaURL == "" {
+		kafkaURL = "localhost:9092"
+	}
+	var kafkaTopic string
+	if kafkaTopic = os.Getenv("EVENTS_TOPIC"); kafkaTopic == "" {
+		kafkaTopic = "test-topic"
+	}
+	seeds := []string{kafkaURL}
+	ctx := context.Background()
+
+	// Setup kafka consumer
+	cl, err := kgo.NewClient(
+		kgo.SeedBrokers(seeds...),
+		kgo.ConsumerGroup("default"),
+		kgo.ConsumeTopics(kafkaTopic),
+		kgo.DisableAutoCommit(),
+	)
+	if err != nil {
+		panic(err)
+	}
+	defer cl.Close()
+
+	var dbURL string
+	if dbURL = os.Getenv("DATABASE_URL"); dbURL == "" {
```

Comment:
Using https://github.com/spf13/viper would make it easier to define these values in a single place. Also wondering if it might make sense to separate the logic to read from config do a different module/getting function like `config.go`